nixos/modules/services/monitoring/grafana-agent.nix

```diff
@@ -37,11 +32,22 @@ in
       };
     };
 
+    extraFlags = mkOption {
+      type = with types; listOf str;
+      default = [ "-config.expand-env" ];
```

Comment:
We want to make sure that `-config.expand-env` stays around even if the user provides a value because it's used to inject the $STATE_DIRECTORY below. Either keep it in the default command-line, or set it with in the config section so it sticks around.